999-SAPonAzure/Coach/01-SAP-Auto-Deployment.md

```diff
@@ -0,0 +1,47 @@
+# Challenge 1: Coach's Guide - SAP Auto Deployment
+
+[< Previous Challenge](./00-prereqs.md) - **[Home](README.md)** - [Next Challenge >](./02-Azure-Monitor.md)
+
+# Notes & Guidance
+
+This challenge introduces an automation tool for SAP Basis persons on provision of (1) all necessary Azure infrastructure (compute/storage/network) (2) server configurations and SAP system installation. The tool is based on Terraform and Ansible script.
+
+## Description
+
+During the exercise, participants will be able to provision a landscape into Azure for SAP environment and then build a fresh SAP S4H system by using an existing backup files into this environment as shown in the following diagram. SAP HANA DB will use Azure Netapp Filesystem for storage volume. 
+
+## Prerequisite
+
+A valid Azure subscription which can create azure resources (computed D-series, E-series, storage, Network..) for Azure regions: uswest2 and useast2
+
+Azure subscription whitelisted with Azure Netapp Filesystem (ANF)
+
+An estimate of an additional $100/daily Azure budget during the challenge-session days.
+ 
+## Success Criteria
+- Complete build of SAP S4H and SAP HANA database on Azure Cloud.
+- Successful Installation of SAP GUI and test logon to SAP Application Server
+
+## Tips
+ 
+For step 5. Make sure that the students create an ubuntu box with a named user "azureuser"
+
+For step 6. Make sure that the students run the commands under named user 'azureuser'
```

Comment:
No, they are scripts to provision SAP system infrastructure (servers, storage, network), SAP GUI, SAP HANA Studio installation files

Reply:
that makes sense, we will call it out in the coach's guide. thanks.

Reply:
updated.

Reply:
So are the Terraform/Ansible files the ones not able to be published?  Are they available online from some other source we could link to?

Reply:
No, we will have to provide deployment scripts to coach's. The deployment scripts contain path to software components, that can not be shared freely.

Reply:
Guys I am VERY confused here... .let's please reset and explain from scratch.

ISSUE: There are no files in Student/Resources even though the students will need to be given files to complete Challenge 1.

SOLUTION: Add the terraform and ansible scripts to Students/Resources and explain to student that the files will be given to them inside of a zip file in the Files tab of the General channel in Teams.

Now Young has said that you CAN NOT put these terraform and ansible files into the github repo because they are NDA/Proprietary and this is the part that confuses me BIG TIME.

What SPECIFICALLY is NDA? Have you confirmed with the product group that this can not be given publicly? Who is allowed to have that information? is it because it is in private preview? When will the restriction be lifted?

There is NO WAY entire Terraform and/or Ansible scripts need to be kept secret. What I suspect is happening is that maybe there is a secret key or a special URL in the script that needs to be hidden? If that's the case then we need to simply remove it from the scripts and replace it with "<Place secret key/url here>". And then in the coach's guide, we need to give them instructions on where to go to get the secrets from the product group.

I hope I'm understanding this correctly, if not, then let's get a group text going and we'll hash it out quick instead of more back and forth in here.

Reply:
the terraform and ansible scripts, contain path to SAP software, downloaded by msft IT which can not be shared.

Reply:
Ok understood and agreed as per our IMs on Wed.

Please slight change the verbiage to tell the students, "The files for this challenge can be found in a zip file in the Files tab of the General Channel for this hack". That is all the students need to know.

In the C1 coach guide put: "As a coach you wil be given files that are not checked into this repo that need to be given to students. When you obtain these files, zip them up into one file and upload them to the Files tab of the General Channel for this hack".

As you ok with those additions, D?